rbd/diff_iterate_by_id.go

```diff
@@ -0,0 +1,93 @@
+//go:build ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#undef _GNU_SOURCE
+#include <errno.h>
+#include <stdlib.h>
+#include <rbd/librbd.h>
+
+// inline wrapper to cast uintptr_t to void*
+static inline int wrap_rbd_diff_iterate3(rbd_image_t image,
+	uint64_t from_snap_id, uint64_t ofs, uint64_t len, uint8_t include_parent,
+	uint8_t whole_object, uintptr_t arg) {
+		return rbd_diff_iterate3(image, from_snap_id, ofs, len, include_parent,
+			whole_object, (void*)diffIterateCallback, (void*)arg);
+};
+*/
+import "C"
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ceph/go-ceph/internal/dlsym"
+)
+
+var (
+	diffIterateByIDOnce sync.Once
+	diffIterateByIdErr  error
+)
+
+// DiffIterateByIDConfig is used to define the parameters of a DiffIterateByID call.
+// Callback, Offset, and Length should always be specified when passed to
+// DiffIterateByID. The other values are optional.
+type DiffIterateByIDConfig struct {
+	SnapID        uint64
```

Comment:
Nit: To avoid potential confusion as to whether it's the "from" snapshot or the "end" snapshot (although I see that the existing binding uses `SnapName` instead of `FromSnapName` too, so you might prefer consistency here):
```suggestion
	FromSnapID        uint64
```

Reply:
Changed, thanks!